### ISSUE TITLE
Configure or config?

### ISSUE THREAD

**Issue by levitte** [openssl/openssl#11218]
As indicated by #11011, but also other comments over time (too old to dig out now), there's a slow ongoing effort to merge `config` and `Configure` into one.

The `config` script is and has always been a wrapper of `Configure` on Unix, while `Configure` was called directly on others.

When they get merged (or one becomes a module that the other calls), I assume that the ultimate end game is that only one of the scripts remain.  In that case, I wonder which name the remaining one should have?  On the one hand, `config` has always been the one to guess the config target, while `Configure` has always been the workhorse that does the rest.

Considering that GNU auto tools commonly create a `configure`, our `Configure` is easily mixed up with it (in fact, that's been the subject of a number of rather confused issues), so with that in mind, it would be sensible to choose `config`, that makes it more different.

Thoughts?

**Comment by levitte** [comment]
In the early development of said effort, the choice made here will decide if `Configure` ends up being a wrapper around `config` or the other way around.  We do want to keep both files initially.

**Comment by t-j-h** [comment]
It is Configure, there is no need to rename it to add to confusion. Keep both as effectively config was traditionally just a simple guess as to the platform - Configure always did the work - and it is very common to bypass 'config' completely so picking that isn't a good choice. There is little benefit in removing one or the other.

People can and do figure out the difference between ./Configure and ./configure pretty quickly so that really isn't an argument to choose 'config' over 'Configure'. 

Given I created 'config' as a **very simple script** back in 1997 to call Configure (for SSLeay-0.9.0) after doing the equivalent of Apache GuessOS it was then a very trivial mapping of platform names into SSLeay platform terms), I think it should be subsumed into Configure.  And at the time all Configure handled was basically which ASM substitutions to make in the various makefiles and header files and which flags picked the "best" outcome for non-asm platforms.

**Comment by levitte** [comment]
> Given I created 'config' as a very simple script back in 1997 to call Configure (for SSLeay-0.9.0) after doing the equivalent of Apache GuessOS it was then a very trivial mapping of platform names into SSLeay platform terms), I think it should be subsumed into Configure.

That's exactly what I mean by merging the two.  My objective here is to incorporate the config target guessing into Configure after the command line arguments are processed, and only if no config target was given on the command line.

With that, `config` would end up being this:

```
#! /bin/sh

THERE=$(dirname $0)
exec $THERE/Configure "$@"
```

> It is Configure, there is no need to rename it to add to confusion.

Did you understand what kind of confusion I was talking about?

**Comment by richsalz** [comment]
I think you need both for this release. You can put a warning message out on the one that is going away for now.

**Comment by levitte** [comment]
Yes, for the upcoming release, they should both remain.  Of course.  I'm thinking further into the future, but the decision made *now* will decide if I end up making `config` the main script and `Configure` just a wrapper script, or the other way around.

**Comment by nhorman** [comment]
given that config was made a wrapper as per https://github.com/openssl/openssl/pull/11230

is this now an action to just remove the config script (and the associated uses in our CI)?

Or should this just be closed @levitte

**Comment by levitte** [comment]
Personally, I'd like to remove the `config` wrapper...  but I'm quite sure there are those who will scream bloody murder 😕

**Comment by nhorman** [comment]
bloody murder sounds like code for 4.0 material to me :)

**Comment by levitte** [comment]
... and I can live with that 😉

**Comment by t8m** [comment]
I do not think there is a point to removing the alias. Closing the issue.
